src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -2153,6 +2155,399 @@ impl Coordinator {
         };
     }
 
+    #[instrument]
+    pub(super) async fn sequence_explain_analyze(
+        &mut self,
+        mut ctx: ExecuteContext,
+        plan: plan::ExplainAnalyzePlan,
+        // TODO: What is this used for? Should we use it for anything? PLAN INSIGHTS seems maybe
+        // correct? But the others could be a no because we always want the introspection queries to
+        // use the auto route cluster?
```

Comment:
Yes, the introspection queries that we run as part of the analysis should run on the introspection cluster (to use the indexes there, without which performance might be bad), but the optimization of the query that is being analyzed should be run on the cluster that would normally run the query.